src/cleanup.c

```diff
@@ -24,3 +24,22 @@ void	ft_free_char_arr(char **array)
 	}
 	free (array);
 }
+
+void	ft_free_mlx(void *mlx_ptr, void *win_ptr, void *img_ptr)
+{
+	if (img_ptr)
+		mlx_destroy_image(mlx_ptr, img_ptr);
+	if (win_ptr)
+		mlx_destroy_window(mlx_ptr, win_ptr);
+	if (mlx_ptr)
```

Comment:
Shouldn't that check encompass the above? As the mlx_ptr is a prerequisite for both functions. Otherwise an error could occur.

Sth like

if (mlx_ptr)
	{
		if (win_ptr)
		{			
			mlx_destroy_window(mlx_ptr, win_ptr);
			win_ptr = NULL;
		}
                if (img_ptr)
		{			
			mlx_destroy_image(mlx_ptr, img_ptr);
			img_ptr = NULL;
		}
		mlx_destroy_display(mlx_ptr);
		free(mlx_ptr);
	}

Reply:
If mlx_ptr doesn't exist, then win_ptr and img_ptr cannot exist since the creating functions need to be passed a mlx_ptr.
So the error cases are
- mlx_ptr was free'd and set to NULL before freeing the other stuff -> we won't do that as evidenced by this function. 
- mlx_ptr was "lost" -> at this point we would crash anyways, before we could free something.

Reply:
You never know what a handy user could do ;) sneakily freeing mlx_ptr while prog is running for example... would always go with the safer version. It's not like we're short on lines here

Reply:
How would the user do that? :o 

And even if he can: is it really expected, that a program does not crash if someone randomly sets a pointer to NULL? Following this argument: Shouldn't we then check every time we use a allocated ressource, since a handy user could set it to NULL? What is the difference with ressources allocated by mlx?

The function gets called while init_mlx and after loop end.
While init_mlx the function call gets filled progressively: 
- if mlx_ptr is success, and win_ptr fails, the mlx_ptr is passed and the other two args are set to NULL
- If mlx_ptr and win_ptr is success, but img ptr fails we pass two pointers, set one to NULL. 
- if all three are success we are happy and don't free anything, since we can get to work.

The function gets called again right after loop end to clean up. While looping we need the mlx_ptr to maintain the XServer connection, else we crash. After loop end we free the three ressources. Since all of them are needed for our program to enter the loop, all of them should be there.

I don't think it introduces more safety, but it just a more contrieved way of doing the same thing.

Reply:
Changed to suggested version